### PR TITLE
feat(cypress): add iatapay connector

### PR DESCRIPTION
**Pull request by pixincreate**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [x] CI/CD

## Description

This PR adds Iatapay connector

- Payment Method: UPI
- Payment Method Types:
  - UPI Collect
  - UPI Intent

Will consider adding `UPI Payments -- Hyperswitch Stripe Compatibility Layer` in future.

Notable changes:

- Removed `Response`s from `Common.js` to accommodate place for `Iatapay`. This does not affect other connectors as it is handled already in `<connector>.js` files.
- PM lists have been refactored to follow alphabetical order to help improve readibility
- During redirection flow in:
	-  `UPI_Collect`, we wait 14 (set by `Iatapay`) + 6 (for page to load) seconds to get the payment to succeed
	- `UPI_Intent`, validate the JSON response, only forward flow
- Lints

<strike>Addition of Iatapay proved that the current implementation of `Commons.js` is just useless. My thought on this is to have additional parameter in every payment_method of `Commons.js` that is checked every time before the test is run. If a connector is not supported, the entire payment method can be ignored.</strike> #5128

In addition to this, `refundPayment` has been modified a bit where I moved all 3DS tests within a describe to make them more intuitive (the split is now made clearer by marking 3DS payments as 3DS payments).

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

NIL

## How did you test it?

Executed `sh run_all.sh --parallel`

|Adyen|Bank of America|Bluesnap|Cybersource|Iatapay|
|-|-|-|-|-|
|image|image|image|image|image|
|NMI|PayPal|Stripe|Trustpay||
|image|image|image|image||

`00018-MandatesUsingPMID` is failing for some connectors. Will be taken up by @likhinbopanna

Iatapay `Upi_intent` is failing during redirection as connector is returning 5xx:

image

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: cypress-tests/cypress/e2e/PaymentTest/00019-UPI.cy.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+import State from "../../utils/State";
+import getConnectorDetails, * as utils from "../PaymentUtils/utils";
+
+let globalState;
+
+describe("UPI Payments - Hyperswitch", () => {
+  let should_continue = true; // variable that will be used to skip tests if a previous test fails
+
+  context("[Payment] [UPI - UPI Collect] Create & Confirm + Refund", () => {
+    should_continue = true; // variable that will be used to skip tests if a previous test fails
```
Comment (Sakilmostak) [inline cypress-tests/cypress/e2e/PaymentTest/00019-UPI.cy.js:13]: remove this since should_continue is already set true for the first case

File: cypress-tests/cypress/e2e/PaymentUtils/Adyen.js
Comment (likhinbopanna) [inline cypress-tests/cypress/e2e/PaymentUtils/Adyen.js:1]: Why did you add this if UPI is not supported by adyen

Comment (likhinbopanna) [inline cypress-tests/cypress/e2e/PaymentUtils/Adyen.js:1]: Why did you add this if UPI is not supported by adyen?

File: cypress-tests/cypress/e2e/PaymentTest/00019-UPI.cy.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+import State from "../../utils/State";
+import getConnectorDetails, * as utils from "../PaymentUtils/utils";
+
+let globalState;
+
+describe("UPI Payments - Hyperswitch", () => {
+  let should_continue = true; // variable that will be used to skip tests if a previous test fails
+
+  context("[Payment] [UPI - UPI Collect] Create & Confirm + Refund", () => {
+    before("seed global state", () => {
```
[the remark this answers is not in the document]
Comment (pixincreate) [reply cypress-tests/cypress/e2e/PaymentTest/00019-UPI.cy.js:13]: We need hooks to be added at context level just because the State needs to be updated prior to upcoming context. Adding that at describe level will not run before each context. We cannot use `beforeEach` here either as it will run before every `it()` resulting in tests to fail.

**Review by likhinbopanna — APPROVED** [review]

File: cypress-tests/cypress/support/redirectionHandler.js [hunk 3]
```diff
@@ -294,7 +250,111 @@ function bankRedirectRedirection(
...
+      case "upi_collect":
+        cy.visit(redirection_url.href);
+        cy.wait(WAIT_TIME_IATAPAY).then(() => {
+          verifyUrl = true;
+        });
+        break;
+      case "upi_intent":
+        cy.request(redirection_url.href).then((response) => {
+          expect(response.status).to.eq(200);
+          console.log(response.body);
```
Comment (Sakilmostak) [inline cypress-tests/cypress/support/redirectionHandler.js:336]: remove this console.log if not required

**Review by Sakilmostak — CHANGES_REQUESTED** [review]

**Review by Sakilmostak — APPROVED** [review]

**Review by Gnanasundari24 — APPROVED** [review]
